src/VisualStudio/IntegrationTests/CSharp/CSharpAutomaticBraceCompletion.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Roslyn.VisualStudio.IntegrationTests;
+using Roslyn.VisualStudio.Test.Utilities;
+using Xunit;
+
+namespace Roslyn.VisualStudio.CSharp.IntegrationTests
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpAutomaticBraceCompletion : IDisposable
+    {
+        private readonly VisualStudioInstanceContext _visualStudio;
+        private readonly Workspace _workspace;
+        private readonly Solution _solution;
+        private readonly Project _project;
+        private readonly EditorWindow _editorWindow;
+
+        public CSharpAutomaticBraceCompletion(VisualStudioInstanceFactory instanceFactory)
+        {
+            _visualStudio = instanceFactory.GetNewOrUsedInstance();
+
+            _solution = _visualStudio.Instance.SolutionExplorer.CreateSolution(nameof(CSharpAutomaticBraceCompletion));
+            _project = _solution.AddProject("TestProj", ProjectTemplate.ClassLibrary, ProjectLanguage.CSharp);
+
+            _workspace = _visualStudio.Instance.Workspace;
+            _workspace.UseSuggestionMode = false;
+
+            _editorWindow = _visualStudio.Instance.EditorWindow;
+        }
+
+        public void Dispose()
+        {
+            _editorWindow.TypeText("\u001B");
+            _visualStudio.Dispose();
+        }
+
+        [Fact]
+        public void BracesInsertionAndTabCompleting()
+        {
+            _editorWindow.Text = @"class C {
+    void Foo() {
+        // Marker
+    }
+}";
+
+            _editorWindow.PlaceCursor("// Marker");
+
+            _editorWindow.TypeText("if (true) {");
+            VerifyEditorTextBeforeAndAfterCursor("        if (true) { ", "}");
+
+            _editorWindow.TypeText("\t");
+            VerifyEditorTextBeforeCursor("        if (true) { }");
+        }
+
+        [Fact]
+        public void BracesOvertyping()
+        {
+            _editorWindow.Text = @"class C {
+    void Foo() {
+        // Marker
+    }
+}";
+
+            _editorWindow.PlaceCursor("// Marker");
+
+            _editorWindow.TypeText("if (true) {");
+
+            _editorWindow.TypeText("}");
+            VerifyEditorTextBeforeCursor("        if (true) { }");
+        }
+
+        [Fact]
+        public void BracesOnReturnNoFormattingOnlyIndentationBeforeCloseBrace()
+        {
+            _editorWindow.Text = @"class C {
+    void Foo() {
+        // Marker
+    }
+}";
+
+            _editorWindow.PlaceCursor("// Marker");
+
+            _editorWindow.TypeText("if (true) {");
+
+            _editorWindow.TypeText("\rvar a = 1;");
```

Comment:
I think `\n` instead of `\r` is more idiomatic. People know what `\n` means, `\r` is much rarer.